dapr/proto/scheduler/scheduler.proto

```diff
@@ -0,0 +1,93 @@
+	// DeleteJob is used by the daprd sidecar to delete a job.
+	rpc DeleteJob(JobRequest) returns (google.protobuf.Empty) {}
+	// GetJob is used by the daprd sidecar to get details of a job.
+	rpc GetJob(JobRequest) returns (GetJobResponse) {}
+	// ListJobs is used by the daprd sidecar to list jobs by app_id.
+	rpc ListJobs(ListJobsRequest) returns (ListJobsResponse) {}
+}
+
+// ConnectClientStream is the message used by the daprd sidecar to connect to the scheduler service.
+// The initiating message has to be a RegisterRequest, followed by a stream of blank health-check messages.
+// The sidecar can send a UnregisterRequest to unregister itself from the scheduler service.
+message ConnectClientStream {
+	// The message type is oneof, so that the sidecar can send a stream of RegisterRequest or UnregisterRequest 
+	// or blank health-check messages.
+	oneof message {
+		RegisterRequest register = 1;
+		UnregisterRequest unregister = 2;
+	}
+}
+
+message ConnectServerStream {
+	// It is empty as of now. A message to be sent here to act as a response to client stream.
+	// This acts as a health check for the server stream.
+}
+
+message RegisterRequest {
+		// The id of the application (app_id). This is used as the key in etcd for the Scheduler Service.
+		string app_id = 1;
+		// The namespace.
+		string namespace = 2;
+		// The hostname of the daprd sidecar.
+		string hostname = 3;
+		// The port of the daprd sidecar.
+		int32 port = 4;
+}
+
+// UnregisterRequest is the message used by the daprd sidecar to unregister itself from the scheduler service.
+message UnregisterRequest {
+	// Job name to be unregistered.
+	string job_name = 1;
+}
+
+// ScheduledJob is the message used by the daprd sidecar to schedule a job.
+message ScheduledJob {
+	// The job to be scheduled.
+	runtime.v1.Job job = 1;
+
+	// Namespace of the job
+	string namespace = 2;
+
+	// The metadata associated with the job.
+	// The sidecar will create the unique `key` for storing data in the state store and pass the generated `key` along to the scheduler service for data lookup upon ‘trigger’ time later on. 
+	// The sidecar will also add metadata in order to know whether this job is registered for an actor. This is needed, as the routing mechanism for actors is different for the callback.
+	map<string,string> metadata = 3;
+}
+
+// JobRequest is the message used by the daprd sidecar to delete or get a job.
+message JobRequest {
+	string job_name = 1;
+}
+
+// GetJobResponse is the response message to convey the details of a job.
+message GetJobResponse {
+	ScheduledJob job = 1;
+}
+
+// ListJobsRequest is the message to list jobs by app_id.
+message ListJobsRequest {
+    // The id of the application (app_id) for which to list jobs.
+    string app_id = 1;
+}
+// ListJobsResponse is the response message to convey the list of jobs.
+message ListJobsResponse {
+    // List of jobs that match the request criteria.
+    repeated ScheduledJob jobs = 1;
+}
+
```

Comment:
why are these different from `runtime.v1.Job` ? Why is `ScheduledJob` being used here?

Reply:
For Scheduler Service, `ScheduledJob` is used consistently for rpc's here.
Is your point that we might not need `metadata` here, as a part of this response?

Reply:
I think it's a bit confusing as is. My understanding is that we should have a: `ScheduleJobRequest` which includes a `Job` to avoid confusion. and the `ListJobResponse` should include `repeated Job jobs = 1;` Where you need to add the job definition [as defined in the proposal](https://github.com/dapr/proposals/pull/44/files#diff-7db04d746377afee6db345bec89c4198b27d604c1747d6f994d4ac9db28bd500R365), like below where we use the `dapr.proto` job def plus the extras we need internally to include `namespace` && `metadata` both of which are definitely needed: 

```
    // Job is the definition of a job.
    message Job {
        daprv1.Job job = 1; // note to update this ref to be proper
        // Namespace of the job.
        string namespace = 2;
        // The metadata associated with the job.
        // The sidecar will create the unique `key` for storing data in the state store and pass the generated `key` along to the scheduler service for data lookup upon ‘trigger’ time later on. 
        // The sidecar will also add metadata in order to know whether this job is registered for an actor. This is needed, as the routing mechanism for actors is different for the callback.
        map<string,string> metadata = 3;
    }
```